### PR TITLE
fix: Bring back DefaultVerify

### PR DESCRIPTION
**Pull request by NimJay**
* Jump Start Solutions (JSS) use Blueprints/Terraform in the following way:
    * `terraform apply ...` - to deploy the JSS
    * `terraform destroy ...` - to delete the JSS deployment
* In the past, we haven't had a need to test that a second `terraform apply ...` (just after the first `terraform apply`) would work as expected.
* But now, with MiM, users will need to run `terraform apply` a second time.
* So I'm bringing back the `DefaultVerify` test.
* Learn [more about DefaultVerify here](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/2213).

### CODE REVIEW

**Review by apeabody — APPROVED** [review]

**Comment by NimJay** [comment]
Failures seen in integration tests:
```
# module.simple.module.filestore.google_filestore_instance.xwiki must be replaced
resource "google_filestore_instance" "xwiki" {
...
    ~ networks {
...
        ~ network           = "xwiki-gce" -> "projects/ci-migrate-gce-a471/global/networks/projects/ci-migrate-gce-a471/global/networks/xwiki-gce" # forces replacement
...
# module.simple.module.vm.module.instance_template.google_compute_instance_template.tpl must be replaced
resource "google_compute_instance_template" "tpl" {
    ...
    ~ metadata_startup_script = (sensitive value) # forces replacement
```

File: infra/provider.tf [hunk 1]
```diff
@@ -19,7 +19,7 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = "~> 4.84"
+      version = "= 4.28.0"
```
Comment (NimJay) [inline infra/provider.tf:22]: I'm using hashicorp/google Provider version 4.28.0 because that's where the fix initially landed.

Comment (NimJay) [reply infra/provider.tf:22]: Even with hashicorp/google Provider version 4.28.0, we get the "network ... # forces replacement" error.
Relevant [logs here](https://pantheon.corp.google.com/cloud-build/builds/828aaf0d-f337-4da2-b11d-c4efa0c5dd36;step=3?e=13803378&mods=notebooks_use_staging_api&project=cloud-foundation-cicd):
```
- Installing hashicorp/google v4.28.0...
```
```
~ network           = "xwiki-gce" -> "projects/ ... /global/networks/projects/ ... /global/networks/xwiki-gce" # forces replacement
```

File: infra/modules/database/main.tf [hunk 2]
```diff
@@ -72,7 +72,7 @@ resource "google_secret_manager_secret" "sql_password" {
   project   = var.project_id
 
   replication {
-    auto {}
+    automatic = true
```
Comment (NimJay) [inline infra/modules/database/main.tf:75]:
```suggestion
    auto {}
```

File: infra/provider.tf (hunk 1 above)
Comment (NimJay) [inline infra/provider.tf:22]:
```suggestion
      version = "~> 4.84"
```

File: infra/modules/filestore/main.tf [hunk 3]
```diff
@@ -27,4 +27,8 @@ resource "google_filestore_instance" "xwiki" {
     capacity_gb = 1024
     name        = "xwiki_file_share"
   }
+  lifecycle {
+    // See https://github.com/hashicorp/terraform-provider-google/issues/16548
+    ignore_changes = [networks.0.network]
```
Comment (NimJay) [inline infra/modules/filestore/main.tf:32]:
```suggestion
    ignore_changes = [networks.0.network]
```

**Pull request closed without merging**